maths/ugly_numbers.py

```diff
@@ -0,0 +1,48 @@
+"""
+Ugly numbers are numbers whose only prime factors are 2, 3 or 5. The sequence
+1, 2, 3, 4, 5, 6, 8, 9, 10, 12, 15, … shows the first 11 ugly numbers. By convention,
+1 is included.
+Given an integer n, we have to find the nth ugly number.
+
+For more details, refer this article
+https://www.geeksforgeeks.org/ugly-numbers/
+"""
+
+
+def ugly_numbers(n: int) -> int:
+    """
+    Returns the nth ugly number.
+    >>> ugly_numbers(100)
+    1536
+    >>> ugly_numbers(1)
+    1
+    >>> ugly_numbers(20)
+    36

```

Comment:
Please do not just test the "happy path"...  Please add tests for ugly_numbers(-5) and ugly_numbers(5.5).

Reply:
I wrote the tests